### ISSUE TITLE
[enhancement] Implementation of a "description" field for each video in a playlist.

### ISSUE THREAD

**Issue by JunioCalu** [ffplayout/ffplayout#622]
Implement a "description" field for each video in the playlist, accompanied by a checkbox, and a button to save the description to disk. If the checkbox is enabled, a textual description of the current video should be automatically displayed (subtitled) on the screen via the zmq messaging protocol (libzmq). This would be useful in cases where old recordings need to be replayed, but much of the audience may not realize that these are reruns and need to be informed about the replay to avoid confusion over treating old quotes and information spoken in the video as current. A typical user could add the following description (subtitle) to a video: "This video is a replay of a program recorded on 04/26/2020."

**Comment by jb-alvarado** [comment]
Some thoughts about it:

- To me a field name description is not fitting exactly, maybe subtitle or something else fits better.
- Why needing libzmq? That would involved another program like ffpapi for that work, which is not convenient for me. 
- when is just a drawtext filter statement, why not using just custom filter parameter for that?

**Comment by JunioCalu** [comment]
> * To me a field name description is not fitting exactly, maybe subtitle or something else fits better.
> * Why needing libzmq? That would involved another program like ffpapi for that work, which is not convenient for me.
> * when is just a drawtext filter statement, why not using just custom filter parameter for that?

Suppose a 24-hour programming schedule consists of 100 videos, all of which are replays of old recordings, each recorded on different dates. If a TV operator needs to add a long and completely different description (caption) to each video, how can this captioning process be automated? Conventionally, the TV operator would have to monitor the broadcast in real-time and manually add a new message (caption) each time a new clip is displayed through the "message" menu (or page) of the ffplayout. Another option would be to rename the title or filename according to the description (caption) and implement a Python script to extract this title or filename and send it as a "message" (drawtext caption) to the zmq-enabled port on the ffplayout. This automation solution could be problematic because, in cases where not all videos require captions, it would be challenging to determine which title or filename should be extracted and converted into a zmq message to be displayed on ffplayout. Adding some flag to the filename could be a potential solution, but it would not be a standardized solution and could lead to failures since it would depend on human action and attention.

I think it would be feasible to use the existing presets on the ffplayout's "message" settings page. A dropdown menu could be added that accesses the predefined and user-saved message settings (drawtext caption) on the "message" settings page. Then, this new "description field" feature would consist of a dropdown selection menu listing the existing caption presets, a text input field, a checkbox to activate or deactivate the message (drawtext caption) during the clip's broadcast, a cancel button, and a save to disk button.

**Comment by clseibold** [comment]
Yeah, the custom filter would have to be added everytime you create a new playlist/program. Instead, something like this definitely should be automated some way: have ffplayout automatically draw the text introducting the clip's title/filename/description at the start of the clip.
